code/core/src/shared/status-store/index.ts

```diff
@@ -0,0 +1,227 @@
+
+export type UseStatusStore = (selector?: (statuses: StatusesByStoryIdAndTypeId) => any) => any;
+
+export function createStatusStore(params: {
+  universalStatusStore: UniversalStore<StatusesByStoryIdAndTypeId, StatusStoreEvent>;
+  useUniversalStore?: never;
+  environment: StatusStoreEnvironment;
+}): {
+  getStatusStoreByTypeId: (typeId: StatusTypeId) => StatusStoreByTypeId;
+  fullStatusStore: FullStatusStore;
+};
+export function createStatusStore(params: {
+  universalStatusStore: UniversalStore<StatusesByStoryIdAndTypeId, StatusStoreEvent>;
+  useUniversalStore: typeof managerUseUniversalStore;
+  environment: StatusStoreEnvironment;
+}): {
+  getStatusStoreByTypeId: (typeId: StatusTypeId) => StatusStoreByTypeId;
+  fullStatusStore: FullStatusStore;
+  useStatusStore: UseStatusStore;
+};
+export function createStatusStore({
+  universalStatusStore,
+  useUniversalStore,
+  environment,
+}: {
+  universalStatusStore: UniversalStore<StatusesByStoryIdAndTypeId, StatusStoreEvent>;
+  useUniversalStore?: typeof managerUseUniversalStore;
+  environment: StatusStoreEnvironment;
+}): {
+  getStatusStoreByTypeId: (typeId: StatusTypeId) => StatusStoreByTypeId;
+  fullStatusStore: FullStatusStore;
+  useStatusStore?: UseStatusStore;
+} {
+  const fullStatusStore: FullStatusStore = {
+    getAll() {
+      return universalStatusStore.getState();
+    },
+    set(statuses) {
+      universalStatusStore.setState((state) => {
+        // Create a new state object to merge with the current state
+        const newState = { ...state };
+
+        // Process each status and merge it into the appropriate storyId record
+        for (const status of statuses) {
+          const { storyId, typeId } = status;
+
+          newState[storyId] = { ...(newState[storyId] ?? {}), [typeId]: status };
+        }
+        return newState;
+      });
+    },
+    onAllStatusChange(
+      listener: (
+        statuses: StatusesByStoryIdAndTypeId,
+        prevStatuses: StatusesByStoryIdAndTypeId
+      ) => void
+    ): ReturnType<typeof universalStatusStore.onStateChange> {
+      return universalStatusStore.onStateChange((state, prevState) => {
+        listener(state, prevState);
+      });
+    },
+    onSelect(listener) {
+      return universalStatusStore.subscribe(StatusStoreEventType.SELECT, (event) => {
+        listener(event.payload);
+      });
+    },
+    selectStatuses: (statuses: Status[]) => {
+      universalStatusStore.send({ type: StatusStoreEventType.SELECT, payload: statuses });
+    },
+    unset(storyIds?: StoryId[]): void {
+      // If no storyIds are provided, remove all statuses
+      if (!storyIds) {
+        universalStatusStore.setState({});
+        return;
+      }
+
+      universalStatusStore.setState((state) => {
+        const newState = { ...state };
+        for (const storyId in newState) {
+          if (storyIds.includes(storyId)) {
+            delete newState[storyId];
+          }
+        }
```

Comment:
Why iterate every state value rather than the (likely shorter) array of `storyIds`?